torch/csrc/profiler/collection.cpp

```diff
@@ -33,11 +33,18 @@ RawTensorMetadataBase::RawTensorMetadataBase(const at::Tensor& t)
     : data_{t.has_storage() ? t.storage().data() : nullptr},
       dtype_{t.scalar_type()},
       layout_{t.layout()},
-      dim_{static_cast<uint32_t>(t.sizes().size())} {
+      size_dim_{static_cast<uint32_t>(t.sizes().size())},
+      stride_dim_{static_cast<uint32_t>(t.strides().size())} {
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
       t.sizes().size() <= std::numeric_limits<uint32_t>::max(),
       "Cannot profile Tensors of size > uint32 max. Got dim: ",
       t.sizes().size());
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+      t.sizes().size() != t.strides().size(),
+      "Tensor has mismatching sizes and strides. Sizes: ",
+      t.sizes(),
+      " Strides: ",
+      t.strides());
```

Comment:
did you mean to print the sizes and strides rather than their dimensions? Seems like we're checking dims here